packages/camera/camera/android/src/main/java/io/flutter/plugins/camera/CameraPermissions.java

```diff
@@ -90,9 +106,9 @@ public boolean onRequestPermissionsResult(int id, String[] permissions, int[] gr
 
       alreadyCalled = true;
       if (grantResults[0] != PackageManager.PERMISSION_GRANTED) {
-        callback.onResult("cameraPermission", "MediaRecorderCamera permission not granted");
+        callback.onResult(CAMERA_ACCESS_DENIED, CAMERA_ACCESS_DENIED_MESSAGE);
       } else if (grantResults.length > 1 && grantResults[1] != PackageManager.PERMISSION_GRANTED) {
-        callback.onResult("cameraPermission", "MediaRecorderAudio permission not granted");
+        callback.onResult(AUDIO_ACCESS_DENIED, AUDIO_ACCESS_DENIED_MESSAGE);
```

Comment:
do we know if grantResults[1] always equals the audio?

Reply:
Peeking at the implementation, I see that the order of `grantResults` mirrors the order in which permissions are requested, and we request them in that order.

Reply:
Well, the [documentation](https://developer.android.com/reference/androidx/core/app/ActivityCompat.OnRequestPermissionsResultCallback#public-methods_1) does note that the results correspond to the list of permissions and another piece of verification is this sample app that they note in the source code as to how to use the method that uses the same method ([link](https://github.com/googlearchive/android-RuntimePermissions/blob/96612da3a0b4489fdf818a624a7e26fc768c65c9/Application/src/main/java/com/example/android/system/runtimepermissions/MainActivity.java#L254)).